### ISSUE TITLE
Line numbers

### ISSUE THREAD

**Issue by tomitrescak** [mazko/jsjavaparser#7]
How difficult would it be to include line numbers in the generated AST? Without line numbers it's not possible to generate source maps. Do you know of any other technique on how those source maps can be generated? Thanks!

**Comment by mazko** [comment]
I don't see easy way to do it - maybe using location() function inside each parser action. I would also try to implement custom PEG plugin which appends location() info to each node automatically.

**Comment by tomitrescak** [comment]
Yes, I solved it by adding location into each function and then recompiled grammar with pegjs, was not that much work. Would you consider PR?

**Comment by mazko** [comment]
Currently I have no much time to support this project, but any pull requests are welcome of course.

**Comment by gorootde** [comment]
@tomitrescak Is there any fork / PR yet? Where exactly did you do your changes to support output of Line numbers?

**Comment by tomitrescak** [comment]
Here you go: https://gist.github.com/tomitrescak/6c18bfb437c123ebcb8bd694a062cad7
But it's probably way behind any changes done since June/2016 in the official package.
